Nos/Service/Keychain.swift

```diff
@@ -50,3 +50,22 @@ enum KeyChain {
         return SecItemDelete(query as CFDictionary)
     }
 }
+
+class MockKeychain: Keychain {
```

Comment:
I think I just realized why I prefer not subclassing for mocks: if we add a new func to Keychain that does something real but forget to override it in the Mock to do something fake, now we have a MockKeychain that does something real. Would a protocol or [the pointfree approach](https://pointfreeco.github.io/swift-dependencies/main/documentation/dependencies/designingdependencies) work, or do those approaches just make everything too messy?

Reply:
Yeah, a protocol is just better. I think I'm still conditioned by the days when generics would just not be usable in random places, but now that we have existential types I don't think that's the case. I will change it.